mono/mini/debugger-agent.c

```diff
@@ -4972,7 +4987,11 @@ debugger_agent_breakpoint_from_context (MonoContext *ctx)
 		return;
 
 	orig_ip = (guint8 *)MONO_CONTEXT_GET_IP (ctx);
+#ifndef __s390x__
 	MONO_CONTEXT_SET_IP (ctx, orig_ip - 1);
+#else
```

Comment:
Why is this needed ?

Reply:
S390x call instructions are 2, 4, or 5 bytes in length. The one used for these calls are 2 bytes.

Reply:
This is supposed to make sure that ip is in the previous instruction, it shouldn't depend on the instruction size, many other arches have 2/4 byte instructions.

Reply:
Are you saying it's not dependent on it being set to the starting address of the instruction just that it's enough to be somewhere in that instruction?

Reply:
Yes.

Reply:
If that doesn't work, there might be an issue somewhere else.

Reply:
I'll build it without that change and see how it goes. If it works for the other archs then I don't see why it won't work here. I just assumed it needed to point at the start of the instruction. What does it use that value for elsewhere in the agent?

Reply:
Mainly to compute which sequence point we are stopped at i think.

Reply:
Pushed change to revert the special case. Tests still pass (bar the same 1 that failed before).